### ISSUE TITLE
per metric mask 

### ISSUE THREAD

**Issue by sdika** [stnava/ANTs#360]
Hi,

unless I missed something, it is not possible now to use one mask per metric (only one mask per stage is possible).
Right know it seems I can specify a single mask that will be applied to all metric and a  metric-wise (but global) weight.

I work on multimodality registration and I know that some of the modalities are better in some known region than other. So I would like to specify a different "weight image" for each metric.

For the command line syntax, the weight parameter that is already present in the metric specification could accept either a scalar or a weight image.

is it possible, to add this feature ?

Michaël

**Comment by ntustison** [comment]
I agree that the proposed modification would be useful for certain cases.  And it's definitely possible.  We can certainly leave this as a 'to-do' item but, honestly, the code overhaul would be quite extensive and so I don't foresee it getting done anytime soon.

**Comment by sdika** [comment]
I completely understand
thanks

**Comment by stnava** [comment]
you can do a work around via the following observation :

if the metric is roughly linear then you can apply weights to the images
directly.

e.g.  w * < i - j, i -j > = < w * i - w * j, i - j >

would recommend working out the math but that's the basic idea


brian

**Comment by sdika** [comment]
that indeed works to incorporate the weight in the metric.
However, the nonlinear deformation makes it problematic:
w^2(x) || I(T(x)) - J(x) ||^2 = || w(x)I(T(x)) - w(x)J(x) ||^2

multiplication by w would approximate w(x)I(T(x)) by w(T(x))I(T(x))
for strong deformations, I don't think the approximation would be valid,

thanks anyway
